UnityProject/Assets/Scripts/Util/SweetExtensions.cs

```diff
@@ -537,6 +538,33 @@ public static int RoundToLargestInt(this float source)
 		return Mathf.CeilToInt(source);
 	}
 
+	/// <summary>
+	/// Removes an item from a ConcurrentBag
+	/// Creates CG
+	/// </summary>
+	public static void Remove<T>(this ConcurrentBag<T> source, T toRemove)
+	{
+		if (source.IsEmpty) return;
+
+		//GC's oh well
+		var temp = new List<T>(source.Count - 1);
+
+		//Caution! If there's a thread that is adding stuff constantly to this bag then this wont stop!
+		while (source.IsEmpty == false)
+		{
+			source.TryTake(out var y);
+
+			if (y.Equals(toRemove)) continue;
+
+			temp.Add(y);
+		}
+
+		foreach (var item in temp)
+		{
+			source.Add(item);
+		}
+	}
```

Comment:
Downside this GC's

Reply:
problem is the transfer method, removes all the gas from a gas mix

Reply:
oh yeah that too

Reply:
Might be able to make a custom function for transfer so it doesnt need to loop through all the gases and removing them individually

Reply:
you going to have to use it everywhere though

Reply:
> problem is the transfer method, removes all the gas from a gas mix

Which transfer method, i cant really find one which'll do a lot of removing

Reply:
in CalcMeanGasMix 
meanGasMix.Clear();
and 
	public static GasData CopyTo(this GasData oldData, GasData CopyTo)
	CopyTo.Clear();

Reply:
just to name a few

Reply:
CalcMeanGasMix => doesnt remove the moles so wont trigger the remove all the time (will only trigger if the mean moles gets close to 0)

meanGasMix.Clear(); => doesnt trigger the remove code

public static GasData CopyTo => doesnt trigger the remove code

CopyTo.Clear(); => doesnt trigger the remove code

The remove code is only triggered here:

![image](https://user-images.githubusercontent.com/56727168/142437172-aa4bcc7d-bdb7-46de-9cd8-9ac09d295a5d.png)

Which only happens if the moles is tiny (0.0001)

Reply:
ConcurrentBag surely would have to be cleared if you're clearing the entire mix 
otherwise you would have loads of nitrogen = zero 
oxygen = zero

Reply:
If you are clearing the whole mix then this is called:

![image](https://user-images.githubusercontent.com/56727168/142461702-87baee11-8d03-4f97-8a2d-50d9d685016d.png)

`public static void Remove<T>(this ConcurrentBag<T> source, T toRemove)` is only used to remove specific entries not to clear the whole thing

Reply:
oh ok, so, it's the Remove x  That generates gc ok